### ISSUE TITLE
FEATURE: Add an option to automatically center the viewport when `follow_current_file` option is enabled

### ISSUE THREAD

**Issue by InfiniteRain** [nvim-neo-tree/neo-tree.nvim#1302]
### Did you check the docs?

- [X] I have read all the docs.

### Is your feature request related to a problem? Please describe.

I use neotree with a pretty big project. The problem that I'm experiencing is that with the `follow_current_file` option being enabled, my current selection ALWAYS jumps down when I switch between files via Harpoon or Telescope. I have to manually type `zz` in order to center it manually every time. 
![image](https://github.com/nvim-neo-tree/neo-tree.nvim/assets/19305779/8112e25e-4dfd-47ca-9add-d72ceb4924dd)

### Describe the solution you'd like.

I would like to see an additional option, called something along the lines of `center_on_followed`, into the existing `follow_current_file` configuration. When enabled, it will automatically center on navigation.

Alternatively, a hook could be added that gets run every single time a follow occurs. This way I can configure it to run `zz` in the viewport.

### Additional Context

If the maintainers are too busy to give this request any time, I would be willing to work on this and contribute myself.

**Comment by cseickel** [comment]
Check out `:h neo-tree-events`, I think you can add your `zz` command in a handler for the `after_render` event and it will do what you want.

**Comment by InfiniteRain** [comment]
Executing `zz` on `after_render` doesn't seem to work properly at all. But I found a workaround.

For anyone wondering if it's possible, I achieved the effect by doing the following:

```lua
local execute_win_command = function(winid, cmd)
	if vim.api.nvim_get_current_win() == winid then
		vim.cmd(cmd)
	else
		vim.cmd("call win_execute(" .. winid .. [[, "]] .. cmd .. [[")]])
	end
end

neo_tree.setup({
	-- ...
	event_handlers = {
		{
			event = "neo_tree_window_after_open",
			handler = function(state)
				execute_win_command(state.winid, "setlocal scrolloff=10000")
			end,
		},
	},
	-- ...
})
```

Feel free to close this issue.

**Comment by pysan3** [comment]
If I'm not mistaken I think after this PR: https://github.com/nvim-neo-tree/neo-tree.nvim/pull/1377, it is gonna be centered by default.

**Comment by enter-dragon** [comment]
> If I'm not mistaken I think after this PR: #1377, it is gonna be centered by default.
> 
> Could you check it out @InfiniteRain ?

I'm at 16d1b19 (3.23) and still not centering.

**Comment by pysan3** [comment]
@enter-dragon Thanks for your response. I'll add the `how-to` label then.

**Comment by jondkinney** [comment]
I don't think this solution is ideal. Aside from the issue of it not really being centered unless enough nodes are open above and below the current file, it also means that when you scroll in the tree, your scroll position is kept "centered" until the buffer has scrolled enough to get to the beginning or end of the tree after which point it moves the cursor "normally". 

In an ideal world, IMO, the tree would function normally in terms of direct scroll interaction while the neo-tree buffer is focused and active, and it would jump the position of the open file, centering it (to exactly the same place that zz would in a normal buffer) when navigating splits or opening buffers/files.

I can't speak for everyone, but the reason I want this "follow and keep centered" feature, is so that my eyes will always know where to look to find the open file/buffer in the tree.

Finally, I'm not sure if it's the neo-tree default, or something my distro (AstroNvim) is adding, but I can't even manually do `zz` in neo-tree. The first `z` closes all nodes! 😂
